pkg/util/resource-recommend/resource/k8s_resource.go

```diff
@@ -19,60 +19,42 @@ package resource
 import (
 	"context"
 	"encoding/json"
+	"fmt"
+	v1 "k8s.io/api/apps/v1"
+	appsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
 
 	"github.com/pkg/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	k8stypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/klog/v2"
 	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/kubewharf/katalyst-api/pkg/apis/recommendation/v1alpha1"
 )
 
-func ConvertAndGetResource(ctx context.Context, client k8sclient.Client, namespace string, targetRef v1alpha1.CrossVersionObjectReference) (*unstructured.Unstructured, error) {
-	klog.V(5).Infof("Get resource in", "targetRef", targetRef, "namespace", namespace)
-	obj := &unstructured.Unstructured{}
-	obj.SetAPIVersion(targetRef.APIVersion)
-	obj.SetKind(targetRef.Kind)
-	if err := client.Get(ctx, k8stypes.NamespacedName{Namespace: namespace, Name: targetRef.Name}, obj); err != nil {
+func ConvertAndGetResource(ctx context.Context, client appsv1.AppsV1Interface, namespace string, targetRef v1alpha1.CrossVersionObjectReference) (*v1.Deployment, error) {
```

Comment:
It is recommended to return *unstructured.Unstructured instead of Deployment. Replace Deployment with *unstructured.Unstructured before returning. That way, when you add other kinds, you don't have to change GetAllClaimedContainers